components/cop_codegen/src/rpn_function.rs

```diff
@@ -1,49 +1,319 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
-use super::Result;
 use heck::CamelCase;
 use proc_macro2::{Span, TokenStream};
 use quote::ToTokens;
-use syn::spanned::Spanned;
-use syn::{
-    parse2, parse_str, FnArg, GenericArgument, Ident, ItemFn, Lifetime, LifetimeDef, PathArguments,
-    Type, TypePath,
-};
-
-pub struct RpnFnGenerator {
-    meta: Vec<Ident>,
+use syn::punctuated::Punctuated;
+use syn::*;
+
+/// Parses an attribute like `#[rpn_fn(vargs, capture = [ctx, output_rows])`.
+#[derive(Debug, Default)]
+struct RpnFnAttr {
+    is_varg: bool,
+    captures: Vec<Expr>,
+}
+
+impl parse::Parse for RpnFnAttr {
+    fn parse(input: parse::ParseStream<'_>) -> Result<Self> {
+        let mut is_varg = false;
+        let mut captures = Vec::new();
+
+        let config_items = Punctuated::<Expr, Token![,]>::parse_terminated(input).unwrap();
+        for item in config_items {
+            match item {
+                Expr::Assign(ExprAssign { left, right, .. }) => {
+                    let left_str = format!("{}", (&left).into_token_stream());
+                    match left_str.as_ref() {
+                        "capture" => {
+                            if let Expr::Array(ExprArray { elems, .. }) = *right {
+                                captures = elems.into_iter().collect();
```

Comment:
Should we check the captures? e.g: Duplicated items in `#[rpn_fn(varg, captures=[ctx,ctx,output_rows]` should be removed.

Reply:
It could be a valid syntax if user defined function accept duplicate arguments. The framework won't intervene.